extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.tracking.js

```diff
@@ -1,35 +1,152 @@
 define('wikia.articleVideo.featuredVideo.tracking', [], function () {
+	var state = getDefaultState();
+	var gaCategory = 'featured-video';
+	var playerInstance;
+	var wasAlreadyUnmuted = false;
+	var depth = 0;
+
 	function getDefaultState() {
 		return {
-			wasAlreadyUnmuted: false
+			wasStartTracked: false,
+			progress: {
+				durationTracked: 0,
+				percentTracked: 0
+			}
 		}
 	}
 
 	function track(gaData) {
-		console.log(gaData);
+		if (!gaData.label) {
+			throw new Error('No tracking label provided');
+		}
+
+		var finalGAData = {
+			action: gaData.action || 'click',
+			category: gaCategory,
+			label: gaData.label,
+			//value tracks sound state: 1 for muted, 0 for unmuted
+			value: Number(playerInstance.getMute())
+		};
+
+		// Will be replaced by connecting Internal + GA trackers
+		console.info(gaData);
 	}
 
-	var state = getDefaultState();
+	return function (providedPlayerInstance, willAutoplay, gaCategory) {
+		playerInstance = providedPlayerInstance;
+		gaCategory = gaCategory;
 
-	return function (playerInstance) {
-		playerInstance.on('play', function () {
+		playerInstance.once('ready', function () {
 			track({
-				label: 'featured-video-play'
+				label: 'load',
+				action: 'impression'
+			});
+
+			var relatedPlugin = playerInstance.getPlugin('related');
+
+			relatedPlugin.on('open', function () {
+				track({
+					label: 'recommended-video',
+					action: 'impression'
+				});
+
+				state = getDefaultState();
+			});
+
+			relatedPlugin.on('play', function (data) {
+				depth++;
+
+				var labelPrefix = data.auto ? 'recommended-video-autoplay-' : 'recommended-video-select-';
+
+				track({
+					label: labelPrefix + data.position,
+					action: 'impression'
+				});
+
+				track({
+					label: 'recommended-video-depth' + depth,
+					action: 'impression'
+				});
 			});
 		});
 
+		playerInstance.once('firstFrame', function () {
+			if (!willAutoplay || state.wasStartTracked || depth > 0) {
+				return;
+			}
+
+			track({
+				label: 'autoplay-start',
+				action: 'impression'
+			});
+		});
+
+		playerInstance.on('play', function () {
+			var label = 'play-resumed';
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the declaration of the unused 'label' variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "label" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)